src/core/AvatarClient.ts

```diff
@@ -155,18 +163,19 @@ export class AvatarClient extends HTTPClient {
   private handleDataReceived(data: Uint8Array) {
     const decoder = new TextDecoder();
     const message: ParsedMessage = JSON.parse(decoder.decode(data));
+
     if (message.type === MessageType.State) {
       const isAvatarSpeaking = message.data.state === MessageState.Speaking;
       this.isAvatarSpeaking = isAvatarSpeaking;
-      if (this.onAvatarSpeakingChange) {
-        this.onAvatarSpeakingChange(isAvatarSpeaking);
-      }
+      this.eventEmitter.emit('avatarSpeakingChange', isAvatarSpeaking);
```

Comment:
thought: this is fine if you wanna keep the same functionality just new syntax, but since we're adding a breaking change I'd suggest we change the event from `avatarSpeakingChange` to `stateChanged` and send the message.data.state as the argument. Let the client do whatever they want with it

Reply:
I like `avatarSpeakingChange` to make it super clear what it does. From what Zain asked and I have seen, the only state event we've really needed is whether the avatar is speaking or not.

Reply:
And if we make those changes in the future, we should clean up our API first. For example, we have a loading state that we're not even using. So, I think we should sort out the states in the API before tweaking the SDK.

Reply:
I understand, not going to push for this change, but my proposal would work exactly the same but instead of receiving a boolean true/false they'd get an enum with the avatar state idle/speaking